src/activity_handlers.cpp

```diff
@@ -1902,7 +1902,8 @@ void activity_handlers::train_finish( player_activity *act, player *p )
         const Skill &skill = sk.obj();
         std::string skill_name = skill.name();
         int old_skill_level = p->get_skill_level( sk );
-        p->practice( sk, 100, old_skill_level );
+        int skill_cap = old_skill_level + 2;
+        p->practice( sk, 100, skill_cap );
```

Comment:
There's no need to declare a separate variable that is used only here and nowhere else.
```suggestion
        p->practice( sk, 100, old_skill_level + 2 );
```